service/history/workflow/mutable_state_impl.go

```diff
@@ -3985,6 +3986,37 @@ func (ms *MutableStateImpl) RetryActivity(
 	return enumspb.RETRY_STATE_IN_PROGRESS, nil
 }
 
+func (ms *MutableStateImpl) truncateRetryableActivityFailure(
+	activityFailure *failurepb.Failure,
+) *failurepb.Failure {
+	namespaceName := ms.namespaceEntry.Name().String()
+	failureSize := activityFailure.Size()
+
+	if failureSize <= ms.config.MutableStateActivityFailureSizeLimitWarn(namespaceName) {
+		return activityFailure
+	}
+
+	throttledLogger := log.With(
+		ms.shard.GetThrottledLogger(),
+		tag.WorkflowNamespace(namespaceName),
+		tag.WorkflowID(ms.executionInfo.WorkflowId),
+		tag.WorkflowRunID(ms.executionState.RunId),
+		tag.BlobSize(int64(failureSize)),
+		tag.BlobSizeViolationOperation("RetryActivity"),
+	)
+
+	sizeLimitError := ms.config.MutableStateActivityFailureSizeLimitError(namespaceName)
+	if failureSize <= sizeLimitError {
+		throttledLogger.Warn("Activity failure size exceeds warning limit for mutable state.")
+		return activityFailure
+	}
+
+	throttledLogger.Warn("Activity failure size exceeds error limit for mutable state, truncated.")
+	serverFailure := failure.NewServerFailure(common.FailureReasonFailureExceedsLimit, false)
```

Comment:
Please put a comment here that this nonretryable (false) is not used, and explain why.